### PR TITLE
Validator: Add overload for isStrongPassword with number return type when returnScore option is true

### PR DESCRIPTION
**Pull request by ghost**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: (scroll down to the isStrongPassword row) https://github.com/validatorjs/validator.js#validators

**Chages made:**
- removed redundant ` | undefined` from the fields in `strongPasswordOptions` interface
- specify that when returnScore option is absent or is false, the return type of isStrongPassword is a boolean
- add an overload so that when returnScore is true, the return type is a number
- Added 3 test scenarios: returnScore absent, returnScore false and returnScore true.

### CODE REVIEW

**Comment by jjdive** [comment]
Related but abandoned PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61293

File: types/validator/index.d.ts [hunk 1]
```diff
@@ -1051,20 +1051,25 @@ declare namespace validator {
...
+        minUppercase?: number;
+        minNumbers?: number;
+        minSymbols?: number;
+        returnScore?: boolean;
+        pointsPerUnique?: number;
+        pointsPerRepeat?: number;
+        pointsForContainingLower?: number;
+        pointsForContainingUpper?: number;
+        pointsForContainingNumber?: number;
+        pointsForContainingSymbol?: number;
```
Comment (peterblazejewicz) [inline types/validator/index.d.ts:1065]:
```suggestion
        minLength?: number | undefined;
        minLowercase?: number | undefined;
        minUppercase?: number | undefined;
        minNumbers?: number | undefined;
        minSymbols?: number | undefined;
        returnScore?: boolean | undefined;
        pointsPerUnique?: number | undefined;
        pointsPerRepeat?: number | undefined;
        pointsForContainingLower?: number | undefined;
        pointsForContainingUpper?: number | undefined;
        pointsForContainingNumber?: number | undefined;
        pointsForContainingSymbol?: number | undefined;
```
those were there for TS 4.4. exact optional property types I think:
https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#exact-optional-property-types
and the native package supports strict 'undefined' for options:
https://github.com/validatorjs/validator.js/blob/63b61629187a732c3b3c8d89fe4cacad890cad99/src/lib/util/merge.js#L3

Comment (jjdive) [reply types/validator/index.d.ts:1065]: Yikes! Just learnt something new! Thanks

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@jjdive thanks!

**Comment by jjdive** [comment]
Ready to merge

**Comment by jlowcs** [comment]
Should this change be fixed to `options?: strongPasswordOptions & { returnScore?: false | undefined }`? (optional `returnScore`)

Otherwise I cannot do this because it's expecting a `returnScore` value:
image

**Comment by peterblazejewicz** [comment]
@jlowcs fixing #62060

**Comment by jjdive** [comment]
> Should this change be fixed to `options?: strongPasswordOptions & { returnScore?: false | undefined }`? (optional `returnScore`)
> 
> Otherwise I cannot do this because it's expecting a `returnScore` value: image

Thanks @jlowcs !!!

**Pull request merged**